pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
So... that error we should never be getting.  So something is horribly
broken.  If we broke specifying requests we need to hold the 1.3 release

On Thu, Sep 1, 2016 at 11:29 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/deploy/controller/deployment/controller.go
> <https://github.com/openshift/origin/pull/10761#discussion_r77196983>:
>
> > @@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
> >
> >  	gracePeriod := int64(10)
> >
> > +	// In case the deployment config strategy does not set the resource requests,
> > +	// we default them.
> > +	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
> > +		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
> > +		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
> > +		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
>
> Let's default to these values if we don't run custom deployments
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10761/files/e16582ed14cbad28129667f6726e0926f3b8b204#r77196983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1GlmOIACYI5I9cHA0MCKhWnsWT5ks5qlu9RgaJpZM4JypmU>
> .
>